internal/cmd/volume/backup/list/list.go

```diff
@@ -0,0 +1,194 @@
+			if err != nil {
+				return fmt.Errorf("get backups: %w", err)
+			}
+			if resp.Items == nil || len(*resp.Items) == 0 {
+				projectLabel, err := projectname.GetProjectName(ctx, params.Printer, params.CliVersion, cmd)
+				if err != nil {
+					params.Printer.Debug(print.ErrorLevel, "get project name: %v", err)
+					projectLabel = model.ProjectId
+				}
+				params.Printer.Info("No backups found for project %s\n", projectLabel)
+				return nil
+			}
+			backups := *resp.Items
+
+			// Truncate output
+			if model.Limit != nil && len(backups) > int(*model.Limit) {
+				backups = backups[:*model.Limit]
+			}
+
+			return outputResult(params.Printer, model.OutputFormat, backups)
+		},
+	}
+
+	configureFlags(cmd)
+	return cmd
+}
+
+func configureFlags(cmd *cobra.Command) {
+	cmd.Flags().Int64(limitFlag, 0, "Maximum number of entries to list")
+	cmd.Flags().String(labelSelectorFlag, "", "Filter backups by labels")
+}
+
+func parseInput(p *print.Printer, cmd *cobra.Command) (*inputModel, error) {
+	globalFlags := globalflags.Parse(p, cmd)
+	if globalFlags.ProjectId == "" {
+		return nil, &errors.ProjectIdError{}
+	}
+
+	limit := flags.FlagToInt64Pointer(p, cmd, limitFlag)
+	if limit != nil && *limit < 1 {
+		return nil, &errors.FlagValidationError{
+			Flag:    limitFlag,
+			Details: "must be greater than 0",
+		}
+	}
+
+	labelSelector := flags.FlagToStringPointer(p, cmd, labelSelectorFlag)
+
+	model := inputModel{
+		GlobalFlagModel: globalFlags,
+		Limit:           limit,
+		LabelSelector:   labelSelector,
+	}
+
+	if p.IsVerbosityDebug() {
+		modelStr, err := print.BuildDebugStrFromInputModel(model)
+		if err != nil {
+			p.Debug(print.ErrorLevel, "convert model to string for debugging: %v", err)
+		} else {
+			p.Debug(print.DebugLevel, "parsed input values: %s", modelStr)
+		}
+	}
+
+	return &model, nil
+}
+
+func buildRequest(ctx context.Context, model *inputModel, apiClient *iaas.APIClient) iaas.ApiListBackupsRequest {
+	req := apiClient.ListBackups(ctx, model.ProjectId)
+
+	if model.LabelSelector != nil {
+		req = req.LabelSelector(*model.LabelSelector)
+	}
+
+	return req
+}
+
+func outputResult(p *print.Printer, outputFormat string, backups []iaas.Backup) error {
+	if backups == nil {
+		return fmt.Errorf("backups is empty")
+	}
+
+	switch outputFormat {
+	case print.JSONOutputFormat:
+		details, err := json.MarshalIndent(backups, "", "  ")
+		if err != nil {
+			return fmt.Errorf("marshal backup list: %w", err)
+		}
+		p.Outputln(string(details))
+		return nil
+
+	case print.YAMLOutputFormat:
+		details, err := yaml.MarshalWithOptions(backups, yaml.IndentSequence(true), yaml.UseJSONMarshaler())
+		if err != nil {
+			return fmt.Errorf("marshal backup list: %w", err)
+		}
+		p.Outputln(string(details))
+		return nil
+
+	default:
+		table := tables.NewTable()
+		table.SetHeader("ID", "NAME", "SIZE", "STATUS", "SNAPSHOT ID", "VOLUME ID", "AVAILABILITY ZONE", "LABELS", "CREATED AT", "UPDATED AT")
+
+		for i := range backups {
```

Comment:
```suggestion
		for _, backup := range backups {
```

or did I miss something? :smile: